backends/backends.c

```diff
@@ -354,9 +353,19 @@ void *backends_main(void *ptr) {
             backend_request_formatter = format_dimension_collected_json_plaintext;
         else
             backend_request_formatter = format_dimension_stored_json_plaintext;
+#else
+        error("AWS Kinesis support isn't compiled");
+#endif /* HAVE_KINESIS */
+    }
+    else if (!strcmp(type, "prometheus remote write")) {
+#if HAVE_PROTOBUF
```

Comment:
HAVE_PROTOBUF indicates the presence of protobufs, this should allow setting ENABLE_BACKEND_PROMETHEUS_REMOTE_WRITE that we should use in turn throughout the implementation to enable prometheus related code.
Makes sense, what do you think? I am just thinking it might be confusing, but i am open to understand this approach too

Reply:
Throughout the C code, we usually use defines in form `HAVE_*`. Should we change this approach? The only `ENABLE_*` is `ENABLE_DBENGINE`.

Reply:
ENABLE_DBENGINE originated from autoconf as advised [here](https://autotools.io/autoconf/arguments.html).